integration-cli/docker_cli_help_test.go

```diff
@@ -117,116 +121,133 @@ func (s *DockerSuite) TestHelpTextVerify(c *check.C) {
 				}
 
-				// If its ok w/o any args then try again with an arg
-				if ec == 0 {
-					args = strings.Split(cmd+" badArg", " ")
-					dCmd = exec.Command(dockerBinary, args...)
-					stdout, stderr, ec, err = runCommandWithStdoutStderr(dCmd)
+				if _, ok := noShortUsage[cmd]; !ok {
+					// For each command run it w/o any args. It will either return
+					// valid output or print a short-usage
+					var dCmd *exec.Cmd
+					var stdout, stderr string
+					var args []string
+
+					// skipNoArgs are ones that we don't want to try w/o
+					// any args. Either because it'll hang the test or
+					// lead to incorrect test result (like false negative).
+					// Whatever the reason, skip trying to run w/o args and
+					// jump to trying with a bogus arg.
+					skipNoArgs := map[string]struct{}{
+						"daemon": {},
+						"events": {},
+						"load":   {},
+					}
+
+					ec := 0
+					if _, ok := skipNoArgs[cmd]; !ok {
+						args = strings.Split(cmd, " ")
+						dCmd = exec.Command(dockerBinary, args...)
+						stdout, stderr, ec, err = runCommandWithStdoutStderr(dCmd)
+					}
+
+					// If its ok w/o any args then try again with an arg
+					if ec == 0 {
+						args = strings.Split(cmd+" badArg", " ")
+						dCmd = exec.Command(dockerBinary, args...)
+						stdout, stderr, ec, err = runCommandWithStdoutStderr(dCmd)
+					}
+
+					if len(stdout) != 0 || len(stderr) == 0 || ec == 0 || err == nil {
+						c.Fatalf("Bad output from %q\nstdout:%q\nstderr:%q\nec:%d\nerr:%q", args, stdout, stderr, ec, err)
+					}
+					// Should have just short usage
+					c.Assert(stderr, checker.Contains, "\nUsage:\t", check.Commentf("Missing short usage on %q\n", args))
+					// But shouldn't have full usage
+					c.Assert(stderr, checker.Not(checker.Contains), "--help=false", check.Commentf("Should not have full usage on %q\n", args))
+					c.Assert(stderr, checker.Not(checker.HasSuffix), "\n\n", check.Commentf("Should not have a blank line on %q\n", args))
 				}
 
-				if len(stdout) != 0 || len(stderr) == 0 || ec == 0 || err == nil {
-					c.Fatalf("Bad output from %q\nstdout:%q\nstderr:%q\nec:%d\nerr:%q", args, stdout, stderr, ec, err)
-				}
-				// Should have just short usage
-				c.Assert(stderr, checker.Contains, "\nUsage:\t", check.Commentf("Missing short usage on %q\n", args))
-				// But shouldn't have full usage
-				c.Assert(stderr, checker.Not(checker.Contains), "--help=false", check.Commentf("Should not have full usage on %q\n", args))
-				c.Assert(stderr, checker.Not(checker.HasSuffix), "\n\n", check.Commentf("Should not have a blank line on %q\n", args))
 			}
-
 		}
 
+		// Divide the list of commands to 5 sets and run the func testcommand on these five sets in parallel
+		// to save runtime of test
+		n := len(cmdsToTest) / 5
+		wg.Add(5)
+
+		go testcommand(cmdsToTest[0 : n-1])
+		go testcommand(cmdsToTest[n : 2*n-1])
+		go testcommand(cmdsToTest[2*n : 3*n-1])
+		go testcommand(cmdsToTest[3*n : 4*n-1])
+		go testcommand(cmdsToTest[5*n:])
```

Comment:
@sainath14: The slicing is still wrong. Here's my original comment, which no longer shows up:

> The slicing here doesn't look right. For example, if n is 2, the first slice will be cmdsToTest[0:1] and the next will be cmdsToTest[2:3]. Neither contains cmdsToTest[1].
> 
> I think this might be easier to verify and maintain if it's turned into a loop.